test/helper.js

```diff
@@ -36,9 +36,16 @@ function startClient (proto, options) {
 
     var client = mqtt.connect(protos[proto], options)
 
+    client._subscribe = client.subscribe
     client.subscribe = promisify(client.subscribe)
+
+    client._unsubscribe = client.unsubscribe
     client.unsubscribe = promisify(client.unsubscribe)
+
+    client._publish = client.publish
     client.publish = promisify(client.publish)
+
+    client._end = client.end
     client.end = promisify(client.end)
```

Comment:
This level of monkey patching will make this brittle in the long run. Instead, we should wrap the client and expose a "raw" property with the original object.

Reply:
What about use https://github.com/mqttjs/async-mqtt?